### ISSUE TITLE
The download button appears to be clickable but the actual download always fail

### ISSUE THREAD

**Issue by xiki-tempula** [yi-ji/NeteaseMusicAbroad#24]
My network is connected via ethernet and I tried to execute python NeteaseMusicProxy.py before starting the NeteaseMusic. The NeteaseMusic is 1.5.9 and I'm using python 2.7.
The song appears to be downloadable as the download button is solid black instead of greyish. However, the download will fail immediately. Here is a copy of the log file.

2018-11-26 23:10:19+0000 [-] Log opened.
2018-11-26 23:10:21+0000 [-] NeteaseMusicProxyFactory starting on 32794
2018-11-26 23:10:21+0000 [-] Starting factory <__main__.NeteaseMusicProxyFactory instance at 0x1092608c0>
2018-11-26 23:10:31+0000 [-] ('DEBUG: Abort on request:', 'music.163.com:443')
2018-11-26 23:10:31+0000 [-] ('DEBUG: Abort on request:', 'music.163.com:443')
2018-11-26 23:10:31+0000 [-] ('DEBUG: Abort on request:', 'music.163.com:443')
2018-11-26 23:10:31+0000 [-] ('DEBUG: Abort on request:', 'music.163.com:443')
2018-11-26 23:10:31+0000 [-] ('DEBUG: Abort on request:', 'music.163.com:443')
2018-11-26 23:10:31+0000 [-] ('DEBUG: Abort on request:', 'music.163.com:443')
2018-11-26 23:10:31+0000 [-] ('DEBUG: Abort on request:', 'music.163.com:443')
2018-11-26 23:10:31+0000 [-] ('DEBUG: Abort on request:', 'music.163.com:443')
2018-11-26 23:10:31+0000 [-] ('DEBUG: Abort on request:', 'music.163.com:443')
2018-11-26 23:10:31+0000 [-] http://music.163.com/eapi/push/init
2018-11-26 23:10:31+0000 [-] Starting factory <__main__.NeteaseMusicProxyClientFactory instance at 0x109327e18>
2018-11-26 23:10:31+0000 [-] http://music.163.com/eapi/purchased/oldprotvip/check
2018-11-26 23:10:31+0000 [-] Starting factory <__main__.NeteaseMusicProxyClientFactory instance at 0x109364a28>
2018-11-26 23:10:31+0000 [-] http://music.163.com/eapi/batch
2018-11-26 23:10:31+0000 [-] Starting factory <__main__.NeteaseMusicProxyClientFactory instance at 0x1093703f8>
2018-11-26 23:10:31+0000 [-] http://music.163.com/eapi/batch
2018-11-26 23:10:31+0000 [-] Starting factory <__main__.NeteaseMusicProxyClientFactory instance at 0x1093730e0>
2018-11-26 23:10:31+0000 [-] http://music.163.com/eapi/pl/count
2018-11-26 23:10:31+0000 [-] Starting factory <__main__.NeteaseMusicProxyClientFactory instance at 0x109373ea8>
2018-11-26 23:10:31+0000 [-] http://music.163.com/eapi/v3/playlist/detail
2018-11-26 23:10:31+0000 [-] Starting factory <__main__.NeteaseMusicProxyClientFactory instance at 0x10937a950>
2018-11-26 23:10:32+0000 [-] "127.0.0.1" - - [26/Nov/2018:23:10:32 +0000] "POST http://music.163.com/eapi/push/init HTTP/1.1" 200 143 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/605.1.15 (KHTML, like Gecko)"
2018-11-26 23:10:32+0000 [-] [Failure instance: Traceback (failure with no frames): <class 'twisted.internet.error.ConnectionDone'>: Connection was closed cleanly.
2018-11-26 23:10:32+0000 [-] ]
2018-11-26 23:10:32+0000 [-] Stopping factory <__main__.NeteaseMusicProxyClientFactory instance at 0x109327e18>
2018-11-26 23:10:32+0000 [-] http://music.163.com/eapi/copyright/restrict/
2018-11-26 23:10:32+0000 [-] Starting factory <__main__.NeteaseMusicProxyClientFactory instance at 0x109387710>
2018-11-26 23:10:32+0000 [-] "127.0.0.1" - - [26/Nov/2018:23:10:32 +0000] "POST http://music.163.com/eapi/batch HTTP/1.1" 200 172 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/605.1.15 (KHTML, like Gecko)"
2018-11-26 23:10:32+0000 [-] [Failure instance: Traceback (failure with no frames): <class 'twisted.internet.error.ConnectionDone'>: Connection was closed cleanly.
2018-11-26 23:10:32+0000 [-] ]
2018-11-26 23:10:32+0000 [-] Stopping factory <__main__.NeteaseMusicProxyClientFactory instance at 0x1093730e0>
2018-11-26 23:10:32+0000 [-] http://music.163.com/eapi/user/getfollows/91802971
2018-11-26 23:10:32+0000 [-] Starting factory <__main__.NeteaseMusicProxyClientFactory instance at 0x1093934d0>
2018-11-26 23:10:32+0000 [-] "127.0.0.1" - - [26/Nov/2018:23:10:32 +0000] "POST http://music.163.com/eapi/purchased/oldprotvip/check HTTP/1.1" 200 107 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/605.1.15 (KHTML, like Gecko)"
2018-11-26 23:10:32+0000 [-] [Failure instance: Traceback (failure with no frames): <class 'twisted.internet.error.ConnectionDone'>: Connection was closed cleanly.
2018-11-26 23:10:32+0000 [-] ]
2018-11-26 23:10:32+0000 [-] Stopping factory <__main__.NeteaseMusicProxyClientFactory instance at 0x109364a28>
2018-11-26 23:10:32+0000 [-] "127.0.0.1" - - [26/Nov/2018:23:10:32 +0000] "POST http://music.163.com/eapi/pl/count HTTP/1.1" 200 328 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/605.1.15 (KHTML, like Gecko)"
2018-11-26 23:10:32+0000 [-] [Failure instance: Traceback (failure with no frames): <class 'twisted.internet.error.ConnectionDone'>: Connection was closed cleanly.
2018-11-26 23:10:32+0000 [-] ]
2018-11-26 23:10:32+0000 [-] Stopping factory <__main__.NeteaseMusicProxyClientFactory instance at 0x109373ea8>
2018-11-26 23:10:32+0000 [-] {'origin': 'orpheus://orpheus', 'content-length': '775', 'accept-language': 'en-gb', 'connection': 'close', 'accept': '*/*', 'user-agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/605.1.15 (KHTML, like Gecko)', 'host': 'music.163.com', 'cookie': 'appver=1.5.9; os=osx; MUSIC_U=6616d3e129bec853009bf9fb670e99475053be85f4706fa6ed8c4c32ab8a94e9f4b344aea7f22ee6e7d3e2c6a3293dc7ccc61bafba67ef9faf9e62a8590fd08a; deviceId=3E704F7A-10C6-5B51-AB74-0F4F88ABE00E%7CD35C16FD-2B1E-46C4-93D1-904BF8D34430; __csrf=6e90e8970fb1280a15544bb27f84dc74; channel=netease; __remember_me=true; MUSIC_A=7533198002a6ee7e325a0e295912e3b6d92b6ac0764332e7cf1a6266fbf8a2f9ddbb8a567d14c186297aa38a2b79159958a12ea57fbed731785d98c6cec68bbd5d77806ba5493980f73ee1024723d0c83afbfadba2b820a8224d1003ec1675bf54185aa64a408aa2ecd7e79d363a7add1bdbe48c2c785fb5; osver=Version%2010.13.6%20(Build%2017G3025);', 'content-type': 'application/x-www-form-urlencoded', 'accept-encoding': 'gzip, deflate'}
2018-11-26 23:10:32+0000 [-] ('response intercepted: ', '/eapi/v3/playlist/detail')
2018-11-26 23:10:32+0000 [-] "127.0.0.1" - - [26/Nov/2018:23:10:32 +0000] "POST http://music.163.com/eapi/v3/playlist/detail HTTP/1.1" 200 777 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/605.1.15 (KHTML, like Gecko)"
2018-11-26 23:10:32+0000 [-] [Failure instance: Traceback (failure with no frames): <class 'twisted.internet.error.ConnectionDone'>: Connection was closed cleanly.
2018-11-26 23:10:32+0000 [-] ]
2018-11-26 23:10:32+0000 [-] Stopping factory <__main__.NeteaseMusicProxyClientFactory instance at 0x10937a950>
2018-11-26 23:10:32+0000 [-] http://music.163.com/eapi/song/lyric
2018-11-26 23:10:32+0000 [-] Starting factory <__main__.NeteaseMusicProxyClientFactory instance at 0x1093965f0>
2018-11-26 23:10:33+0000 [-] "127.0.0.1" - - [26/Nov/2018:23:10:32 +0000] "POST http://music.163.com/eapi/copyright/restrict/ HTTP/1.1" 200 63 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/605.1.15 (KHTML, like Gecko)"
2018-11-26 23:10:33+0000 [-] [Failure instance: Traceback (failure with no frames): <class 'twisted.internet.error.ConnectionDone'>: Connection was closed cleanly.
2018-11-26 23:10:33+0000 [-] ]
2018-11-26 23:10:33+0000 [-] Stopping factory <__main__.NeteaseMusicProxyClientFactory instance at 0x109387710>
2018-11-26 23:10:33+0000 [-] "127.0.0.1" - - [26/Nov/2018:23:10:32 +0000] "POST http://music.163.com/eapi/batch HTTP/1.1" 200 41372 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/605.1.15 (KHTML, like Gecko)"
2018-11-26 23:10:33+0000 [-] [Failure instance: Traceback (failure with no frames): <class 'twisted.internet.error.ConnectionDone'>: Connection was closed cleanly.
2018-11-26 23:10:33+0000 [-] ]
2018-11-26 23:10:33+0000 [-] Stopping factory <__main__.NeteaseMusicProxyClientFactory instance at 0x1093703f8>
2018-11-26 23:10:33+0000 [-] "127.0.0.1" - - [26/Nov/2018:23:10:32 +0000] "POST http://music.163.com/eapi/user/getfollows/91802971 HTTP/1.1" 200 747 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/605.1.15 (KHTML, like Gecko)"
2018-11-26 23:10:33+0000 [-] [Failure instance: Traceback (failure with no frames): <class 'twisted.internet.error.ConnectionDone'>: Connection was closed cleanly.
2018-11-26 23:10:33+0000 [-] ]
2018-11-26 23:10:33+0000 [-] Stopping factory <__main__.NeteaseMusicProxyClientFactory instance at 0x1093934d0>
2018-11-26 23:10:33+0000 [-] "127.0.0.1" - - [26/Nov/2018:23:10:33 +0000] "POST http://music.163.com/eapi/song/lyric HTTP/1.1" 200 1440 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/605.1.15 (KHTML, like Gecko)"
2018-11-26 23:10:33+0000 [-] [Failure instance: Traceback (failure with no frames): <class 'twisted.internet.error.ConnectionDone'>: Connection was closed cleanly.
2018-11-26 23:10:33+0000 [-] ]
2018-11-26 23:10:33+0000 [-] Stopping factory <__main__.NeteaseMusicProxyClientFactory instance at 0x1093965f0>
2018-11-26 23:10:33+0000 [-] http://music.163.com/eapi/v3/song/detail
2018-11-26 23:10:33+0000 [-] Starting factory <__main__.NeteaseMusicProxyClientFactory instance at 0x10939bb90>
2018-11-26 23:10:33+0000 [-] http://music.163.com/eapi/v1/user/bindings/91802971
2018-11-26 23:10:33+0000 [-] Starting factory <__main__.NeteaseMusicProxyClientFactory instance at 0x10939e3f8>
2018-11-26 23:10:34+0000 [-] "127.0.0.1" - - [26/Nov/2018:23:10:34 +0000] "POST http://music.163.com/eapi/v1/user/bindings/91802971 HTTP/1.1" 200 180 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/605.1.15 (KHTML, like Gecko)"
2018-11-26 23:10:34+0000 [-] [Failure instance: Traceback (failure with no frames): <class 'twisted.internet.error.ConnectionDone'>: Connection was closed cleanly.
2018-11-26 23:10:34+0000 [-] ]
2018-11-26 23:10:34+0000 [-] Stopping factory <__main__.NeteaseMusicProxyClientFactory instance at 0x10939e3f8>
2018-11-26 23:10:34+0000 [-] http://music.163.com/eapi/copyright/pay_fee_message/config
2018-11-26 23:10:34+0000 [-] Starting factory <__main__.NeteaseMusicProxyClientFactory instance at 0x1093a10e0>
2018-11-26 23:10:34+0000 [-] "127.0.0.1" - - [26/Nov/2018:23:10:34 +0000] "POST http://music.163.com/eapi/v3/song/detail HTTP/1.1" 200 7132 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/605.1.15 (KHTML, like Gecko)"
2018-11-26 23:10:34+0000 [-] [Failure instance: Traceback (failure with no frames): <class 'twisted.internet.error.ConnectionDone'>: Connection was closed cleanly.
2018-11-26 23:10:34+0000 [-] ]
2018-11-26 23:10:34+0000 [-] Stopping factory <__main__.NeteaseMusicProxyClientFactory instance at 0x10939bb90>
2018-11-26 23:10:34+0000 [-] http://music.163.com/eapi/login/token/refresh
2018-11-26 23:10:34+0000 [-] Starting factory <__main__.NeteaseMusicProxyClientFactory instance at 0x1093a1c20>
2018-11-26 23:10:35+0000 [-] "127.0.0.1" - - [26/Nov/2018:23:10:34 +0000] "POST http://music.163.com/eapi/copyright/pay_fee_message/config HTTP/1.1" 200 873 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/605.1.15 (KHTML, like Gecko)"
2018-11-26 23:10:35+0000 [-] [Failure instance: Traceback (failure with no frames): <class 'twisted.internet.error.ConnectionDone'>: Connection was closed cleanly.
2018-11-26 23:10:35+0000 [-] ]
2018-11-26 23:10:35+0000 [-] Stopping factory <__main__.NeteaseMusicProxyClientFactory instance at 0x1093a10e0>
2018-11-26 23:10:35+0000 [-] "127.0.0.1" - - [26/Nov/2018:23:10:34 +0000] "POST http://music.163.com/eapi/login/token/refresh HTTP/1.1" 200 12 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/605.1.15 (KHTML, like Gecko)"
2018-11-26 23:10:35+0000 [-] [Failure instance: Traceback (failure with no frames): <class 'twisted.internet.error.ConnectionDone'>: Connection was closed cleanly.
2018-11-26 23:10:35+0000 [-] ]
2018-11-26 23:10:35+0000 [-] Stopping factory <__main__.NeteaseMusicProxyClientFactory instance at 0x1093a1c20>
2018-11-26 23:10:36+0000 [-] http://music.163.com/eapi/pl/count
2018-11-26 23:10:36+0000 [-] Starting factory <__main__.NeteaseMusicProxyClientFactory instance at 0x1093a7c68>
2018-11-26 23:10:36+0000 [-] http://music.163.com/eapi/user/playlist/
2018-11-26 23:10:36+0000 [-] Starting factory <__main__.NeteaseMusicProxyClientFactory instance at 0x1093a1b00>
2018-11-26 23:10:36+0000 [-] http://music.163.com/eapi/user/playlist/
2018-11-26 23:10:36+0000 [-] Starting factory <__main__.NeteaseMusicProxyClientFactory instance at 0x1093a1998>
2018-11-26 23:10:36+0000 [-] http://music.163.com/eapi/cdns
2018-11-26 23:10:36+0000 [-] Starting factory <__main__.NeteaseMusicProxyClientFactory instance at 0x1093a1ea8>
2018-11-26 23:10:36+0000 [-] http://music.163.com/eapi/osx/version
2018-11-26 23:10:36+0000 [-] Starting factory <__main__.NeteaseMusicProxyClientFactory instance at 0x1093a1e18>
2018-11-26 23:10:36+0000 [-] http://music.163.com/eapi/v3/playlist/detail
2018-11-26 23:10:37+0000 [-] Starting factory <__main__.NeteaseMusicProxyClientFactory instance at 0x10939eab8>
2018-11-26 23:10:37+0000 [-] "127.0.0.1" - - [26/Nov/2018:23:10:37 +0000] "POST http://music.163.com/eapi/pl/count HTTP/1.1" 200 329 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/605.1.15 (KHTML, like Gecko)"
2018-11-26 23:10:37+0000 [-] [Failure instance: Traceback (failure with no frames): <class 'twisted.internet.error.ConnectionDone'>: Connection was closed cleanly.
2018-11-26 23:10:37+0000 [-] ]
2018-11-26 23:10:37+0000 [-] Stopping factory <__main__.NeteaseMusicProxyClientFactory instance at 0x1093a7c68>
2018-11-26 23:10:37+0000 [-] "127.0.0.1" - - [26/Nov/2018:23:10:37 +0000] "POST http://music.163.com/eapi/user/playlist/ HTTP/1.1" 200 2540 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/605.1.15 (KHTML, like Gecko)"
2018-11-26 23:10:37+0000 [-] [Failure instance: Traceback (failure with no frames): <class 'twisted.internet.error.ConnectionDone'>: Connection was closed cleanly.
2018-11-26 23:10:37+0000 [-] ]
2018-11-26 23:10:37+0000 [-] Stopping factory <__main__.NeteaseMusicProxyClientFactory instance at 0x1093a1b00>
2018-11-26 23:10:37+0000 [-] "127.0.0.1" - - [26/Nov/2018:23:10:37 +0000] "POST http://music.163.com/eapi/user/playlist/ HTTP/1.1" 200 2480 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/605.1.15 (KHTML, like Gecko)"
2018-11-26 23:10:37+0000 [-] [Failure instance: Traceback (failure with no frames): <class 'twisted.internet.error.ConnectionDone'>: Connection was closed cleanly.
2018-11-26 23:10:37+0000 [-] ]
2018-11-26 23:10:37+0000 [-] Stopping factory <__main__.NeteaseMusicProxyClientFactory instance at 0x1093a1998>
2018-11-26 23:10:37+0000 [-] "127.0.0.1" - - [26/Nov/2018:23:10:37 +0000] "POST http://music.163.com/eapi/osx/version HTTP/1.1" 200 612 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/605.1.15 (KHTML, like Gecko)"
2018-11-26 23:10:37+0000 [-] [Failure instance: Traceback (failure with no frames): <class 'twisted.internet.error.ConnectionDone'>: Connection was closed cleanly.
2018-11-26 23:10:37+0000 [-] ]
2018-11-26 23:10:37+0000 [-] Stopping factory <__main__.NeteaseMusicProxyClientFactory instance at 0x1093a1e18>
2018-11-26 23:10:37+0000 [-] "127.0.0.1" - - [26/Nov/2018:23:10:37 +0000] "POST http://music.163.com/eapi/cdns HTTP/1.1" 200 135 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/605.1.15 (KHTML, like Gecko)"
2018-11-26 23:10:37+0000 [-] [Failure instance: Traceback (failure with no frames): <class 'twisted.internet.error.ConnectionDone'>: Connection was closed cleanly.
2018-11-26 23:10:37+0000 [-] ]
2018-11-26 23:10:37+0000 [-] Stopping factory <__main__.NeteaseMusicProxyClientFactory instance at 0x1093a1ea8>
2018-11-26 23:10:37+0000 [-] http://music.163.com/eapi/v3/playlist/detail
2018-11-26 23:10:37+0000 [-] Starting factory <__main__.NeteaseMusicProxyClientFactory instance at 0x109327320>
2018-11-26 23:10:37+0000 [-] {'origin': 'orpheus://orpheus', 'content-length': '775', 'accept-language': 'en-gb', 'connection': 'close', 'accept': '*/*', 'user-agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/605.1.15 (KHTML, like Gecko)', 'host': 'music.163.com', 'cookie': 'appver=1.5.9; os=osx; MUSIC_U=6616d3e129bec853009bf9fb670e99475053be85f4706fa62c9a95431dd31cdd0ecd736df751fcaa76738b890f539048729e1da9eebfbbd3af9e62a8590fd08a; deviceId=3E704F7A-10C6-5B51-AB74-0F4F88ABE00E%7CD35C16FD-2B1E-46C4-93D1-904BF8D34430; __csrf=172f12e6f5aa50088ae3ab4dc70934ad; channel=netease; __remember_me=true; MUSIC_A=7533198002a6ee7e325a0e295912e3b6d92b6ac0764332e7cf1a6266fbf8a2f9ddbb8a567d14c186297aa38a2b79159958a12ea57fbed731785d98c6cec68bbd5d77806ba5493980f73ee1024723d0c83afbfadba2b820a8224d1003ec1675bf54185aa64a408aa2ecd7e79d363a7add1bdbe48c2c785fb5; osver=Version%2010.13.6%20(Build%2017G3025);', 'content-type': 'application/x-www-form-urlencoded', 'accept-encoding': 'gzip, deflate'}
2018-11-26 23:10:37+0000 [-] ('response intercepted: ', '/eapi/v3/playlist/detail')
2018-11-26 23:10:37+0000 [-] "127.0.0.1" - - [26/Nov/2018:23:10:37 +0000] "POST http://music.163.com/eapi/v3/playlist/detail HTTP/1.1" 200 777 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/605.1.15 (KHTML, like Gecko)"
2018-11-26 23:10:37+0000 [-] [Failure instance: Traceback (failure with no frames): <class 'twisted.internet.error.ConnectionDone'>: Connection was closed cleanly.
2018-11-26 23:10:37+0000 [-] ]
2018-11-26 23:10:37+0000 [-] Stopping factory <__main__.NeteaseMusicProxyClientFactory instance at 0x10939eab8>
2018-11-26 23:10:38+0000 [-] {'origin': 'orpheus://orpheus', 'content-length': '775', 'accept-language': 'en-gb', 'connection': 'close', 'accept': '*/*', 'user-agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/605.1.15 (KHTML, like Gecko)', 'host': 'music.163.com', 'cookie': 'appver=1.5.9; os=osx; MUSIC_U=6616d3e129bec853009bf9fb670e99475053be85f4706fa62c9a95431dd31cdd0ecd736df751fcaa76738b890f539048729e1da9eebfbbd3af9e62a8590fd08a; deviceId=3E704F7A-10C6-5B51-AB74-0F4F88ABE00E%7CD35C16FD-2B1E-46C4-93D1-904BF8D34430; __csrf=172f12e6f5aa50088ae3ab4dc70934ad; channel=netease; __remember_me=true; MUSIC_A=7533198002a6ee7e325a0e295912e3b6d92b6ac0764332e7cf1a6266fbf8a2f9ddbb8a567d14c186297aa38a2b79159958a12ea57fbed731785d98c6cec68bbd5d77806ba5493980f73ee1024723d0c83afbfadba2b820a8224d1003ec1675bf54185aa64a408aa2ecd7e79d363a7add1bdbe48c2c785fb5; osver=Version%2010.13.6%20(Build%2017G3025);', 'content-type': 'application/x-www-form-urlencoded', 'accept-encoding': 'gzip, deflate'}
2018-11-26 23:10:38+0000 [-] ('response intercepted: ', '/eapi/v3/playlist/detail')
2018-11-26 23:10:38+0000 [-] "127.0.0.1" - - [26/Nov/2018:23:10:38 +0000] "POST http://music.163.com/eapi/v3/playlist/detail HTTP/1.1" 200 781 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/605.1.15 (KHTML, like Gecko)"
2018-11-26 23:10:38+0000 [-] [Failure instance: Traceback (failure with no frames): <class 'twisted.internet.error.ConnectionDone'>: Connection was closed cleanly.
2018-11-26 23:10:38+0000 [-] ]
2018-11-26 23:10:38+0000 [-] Stopping factory <__main__.NeteaseMusicProxyClientFactory instance at 0x109327320>
2018-11-26 23:10:45+0000 [-] http://music.163.com/eapi/song/download/limit
2018-11-26 23:10:45+0000 [-] Starting factory <__main__.NeteaseMusicProxyClientFactory instance at 0x1093a73f8>
2018-11-26 23:10:45+0000 [-] http://music.163.com/eapi/song/download/limit
2018-11-26 23:10:45+0000 [-] Starting factory <__main__.NeteaseMusicProxyClientFactory instance at 0x1093ab248>
2018-11-26 23:10:45+0000 [-] http://music.163.com/eapi/song/download/limit
2018-11-26 23:10:45+0000 [-] Starting factory <__main__.NeteaseMusicProxyClientFactory instance at 0x1093abcf8>
2018-11-26 23:10:45+0000 [-] "127.0.0.1" - - [26/Nov/2018:23:10:45 +0000] "POST http://music.163.com/eapi/song/download/limit HTTP/1.1" 200 55 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/605.1.15 (KHTML, like Gecko)"
2018-11-26 23:10:45+0000 [-] [Failure instance: Traceback (failure with no frames): <class 'twisted.internet.error.ConnectionDone'>: Connection was closed cleanly.
2018-11-26 23:10:45+0000 [-] ]
2018-11-26 23:10:45+0000 [-] Stopping factory <__main__.NeteaseMusicProxyClientFactory instance at 0x1093abcf8>
2018-11-26 23:10:45+0000 [-] http://music.163.com/eapi/song/enhance/download/url
2018-11-26 23:10:45+0000 [-] Starting factory <__main__.NeteaseMusicProxyClientFactory instance at 0x1093b3878>
2018-11-26 23:10:45+0000 [-] "127.0.0.1" - - [26/Nov/2018:23:10:45 +0000] "POST http://music.163.com/eapi/song/download/limit HTTP/1.1" 200 55 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/605.1.15 (KHTML, like Gecko)"
2018-11-26 23:10:45+0000 [-] [Failure instance: Traceback (failure with no frames): <class 'twisted.internet.error.ConnectionDone'>: Connection was closed cleanly.
2018-11-26 23:10:45+0000 [-] ]
2018-11-26 23:10:45+0000 [-] Stopping factory <__main__.NeteaseMusicProxyClientFactory instance at 0x1093a73f8>
2018-11-26 23:10:45+0000 [-] http://music.163.com/eapi/song/enhance/download/url
2018-11-26 23:10:45+0000 [-] Starting factory <__main__.NeteaseMusicProxyClientFactory instance at 0x1093b54d0>
2018-11-26 23:10:45+0000 [-] "127.0.0.1" - - [26/Nov/2018:23:10:45 +0000] "POST http://music.163.com/eapi/song/download/limit HTTP/1.1" 200 55 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/605.1.15 (KHTML, like Gecko)"
2018-11-26 23:10:45+0000 [-] [Failure instance: Traceback (failure with no frames): <class 'twisted.internet.error.ConnectionDone'>: Connection was closed cleanly.
2018-11-26 23:10:45+0000 [-] ]
2018-11-26 23:10:45+0000 [-] Stopping factory <__main__.NeteaseMusicProxyClientFactory instance at 0x1093ab248>
2018-11-26 23:10:45+0000 [-] http://music.163.com/eapi/song/enhance/download/url
2018-11-26 23:10:45+0000 [-] Starting factory <__main__.NeteaseMusicProxyClientFactory instance at 0x1093b5f38>
2018-11-26 23:10:46+0000 [-] "127.0.0.1" - - [26/Nov/2018:23:10:45 +0000] "POST http://music.163.com/eapi/song/enhance/download/url HTTP/1.1" 200 238 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/605.1.15 (KHTML, like Gecko)"
2018-11-26 23:10:46+0000 [-] [Failure instance: Traceback (failure with no frames): <class 'twisted.internet.error.ConnectionDone'>: Connection was closed cleanly.
2018-11-26 23:10:46+0000 [-] ]
2018-11-26 23:10:46+0000 [-] Stopping factory <__main__.NeteaseMusicProxyClientFactory instance at 0x1093b54d0>
2018-11-26 23:10:46+0000 [-] http://music.163.com/eapi/song/download/limit
2018-11-26 23:10:46+0000 [-] Starting factory <__main__.NeteaseMusicProxyClientFactory instance at 0x1093bac68>
2018-11-26 23:10:46+0000 [-] "127.0.0.1" - - [26/Nov/2018:23:10:45 +0000] "POST http://music.163.com/eapi/song/enhance/download/url HTTP/1.1" 200 238 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/605.1.15 (KHTML, like Gecko)"
2018-11-26 23:10:46+0000 [-] [Failure instance: Traceback (failure with no frames): <class 'twisted.internet.error.ConnectionDone'>: Connection was closed cleanly.
2018-11-26 23:10:46+0000 [-] ]
2018-11-26 23:10:46+0000 [-] Stopping factory <__main__.NeteaseMusicProxyClientFactory instance at 0x1093b5f38>
2018-11-26 23:10:46+0000 [-] http://music.163.com/eapi/song/download/limit
2018-11-26 23:10:46+0000 [-] Starting factory <__main__.NeteaseMusicProxyClientFactory instance at 0x1093be7e8>
2018-11-26 23:10:46+0000 [-] "127.0.0.1" - - [26/Nov/2018:23:10:46 +0000] "POST http://music.163.com/eapi/song/enhance/download/url HTTP/1.1" 200 238 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/605.1.15 (KHTML, like Gecko)"
2018-11-26 23:10:46+0000 [-] [Failure instance: Traceback (failure with no frames): <class 'twisted.internet.error.ConnectionDone'>: Connection was closed cleanly.
2018-11-26 23:10:46+0000 [-] ]
2018-11-26 23:10:46+0000 [-] Stopping factory <__main__.NeteaseMusicProxyClientFactory instance at 0x1093b3878>
2018-11-26 23:10:46+0000 [-] http://music.163.com/eapi/song/download/limit
2018-11-26 23:10:46+0000 [-] Starting factory <__main__.NeteaseMusicProxyClientFactory instance at 0x1093babd8>
2018-11-26 23:10:46+0000 [-] http://music.163.com/eapi/pl/count
2018-11-26 23:10:46+0000 [-] Starting factory <__main__.NeteaseMusicProxyClientFactory instance at 0x1093b5b00>
2018-11-26 23:10:46+0000 [-] "127.0.0.1" - - [26/Nov/2018:23:10:46 +0000] "POST http://music.163.com/eapi/song/download/limit HTTP/1.1" 200 55 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/605.1.15 (KHTML, like Gecko)"
2018-11-26 23:10:46+0000 [-] [Failure instance: Traceback (failure with no frames): <class 'twisted.internet.error.ConnectionDone'>: Connection was closed cleanly.
2018-11-26 23:10:46+0000 [-] ]
2018-11-26 23:10:46+0000 [-] Stopping factory <__main__.NeteaseMusicProxyClientFactory instance at 0x1093bac68>
2018-11-26 23:10:46+0000 [-] http://music.163.com/eapi/song/enhance/download/url
2018-11-26 23:10:46+0000 [-] Starting factory <__main__.NeteaseMusicProxyClientFactory instance at 0x1093b59e0>
2018-11-26 23:10:46+0000 [-] "127.0.0.1" - - [26/Nov/2018:23:10:46 +0000] "POST http://music.163.com/eapi/song/download/limit HTTP/1.1" 200 55 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/605.1.15 (KHTML, like Gecko)"
2018-11-26 23:10:46+0000 [-] [Failure instance: Traceback (failure with no frames): <class 'twisted.internet.error.ConnectionDone'>: Connection was closed cleanly.
2018-11-26 23:10:46+0000 [-] ]
2018-11-26 23:10:46+0000 [-] Stopping factory <__main__.NeteaseMusicProxyClientFactory instance at 0x1093be7e8>
2018-11-26 23:10:46+0000 [-] http://music.163.com/eapi/song/enhance/download/url
2018-11-26 23:10:46+0000 [-] Starting factory <__main__.NeteaseMusicProxyClientFactory instance at 0x1093ab7e8>
2018-11-26 23:10:46+0000 [-] "127.0.0.1" - - [26/Nov/2018:23:10:46 +0000] "POST http://music.163.com/eapi/song/download/limit HTTP/1.1" 200 55 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/605.1.15 (KHTML, like Gecko)"
2018-11-26 23:10:46+0000 [-] [Failure instance: Traceback (failure with no frames): <class 'twisted.internet.error.ConnectionDone'>: Connection was closed cleanly.
2018-11-26 23:10:46+0000 [-] ]
2018-11-26 23:10:46+0000 [-] Stopping factory <__main__.NeteaseMusicProxyClientFactory instance at 0x1093babd8>
2018-11-26 23:10:46+0000 [-] http://music.163.com/eapi/song/enhance/download/url
2018-11-26 23:10:46+0000 [-] Starting factory <__main__.NeteaseMusicProxyClientFactory instance at 0x1093ab3b0>
2018-11-26 23:10:47+0000 [-] "127.0.0.1" - - [26/Nov/2018:23:10:47 +0000] "POST http://music.163.com/eapi/pl/count HTTP/1.1" 200 327 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/605.1.15 (KHTML, like Gecko)"
2018-11-26 23:10:47+0000 [-] [Failure instance: Traceback (failure with no frames): <class 'twisted.internet.error.ConnectionDone'>: Connection was closed cleanly.
2018-11-26 23:10:47+0000 [-] ]
2018-11-26 23:10:47+0000 [-] Stopping factory <__main__.NeteaseMusicProxyClientFactory instance at 0x1093b5b00>
2018-11-26 23:10:47+0000 [-] "127.0.0.1" - - [26/Nov/2018:23:10:47 +0000] "POST http://music.163.com/eapi/song/enhance/download/url HTTP/1.1" 200 238 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/605.1.15 (KHTML, like Gecko)"
2018-11-26 23:10:47+0000 [-] [Failure instance: Traceback (failure with no frames): <class 'twisted.internet.error.ConnectionDone'>: Connection was closed cleanly.
2018-11-26 23:10:47+0000 [-] ]
2018-11-26 23:10:47+0000 [-] Stopping factory <__main__.NeteaseMusicProxyClientFactory instance at 0x1093b59e0>
2018-11-26 23:10:47+0000 [-] "127.0.0.1" - - [26/Nov/2018:23:10:47 +0000] "POST http://music.163.com/eapi/song/enhance/download/url HTTP/1.1" 200 238 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/605.1.15 (KHTML, like Gecko)"
2018-11-26 23:10:47+0000 [-] [Failure instance: Traceback (failure with no frames): <class 'twisted.internet.error.ConnectionDone'>: Connection was closed cleanly.
2018-11-26 23:10:47+0000 [-] ]
2018-11-26 23:10:47+0000 [-] Stopping factory <__main__.NeteaseMusicProxyClientFactory instance at 0x1093ab7e8>
2018-11-26 23:10:47+0000 [-] http://music.163.com/eapi/song/download/limit
2018-11-26 23:10:47+0000 [-] Starting factory <__main__.NeteaseMusicProxyClientFactory instance at 0x10939b758>
2018-11-26 23:10:47+0000 [-] http://music.163.com/eapi/song/download/limit
2018-11-26 23:10:47+0000 [-] Starting factory <__main__.NeteaseMusicProxyClientFactory instance at 0x10939b170>
2018-11-26 23:10:47+0000 [-] "127.0.0.1" - - [26/Nov/2018:23:10:47 +0000] "POST http://music.163.com/eapi/song/enhance/download/url HTTP/1.1" 200 238 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/605.1.15 (KHTML, like Gecko)"
2018-11-26 23:10:47+0000 [-] [Failure instance: Traceback (failure with no frames): <class 'twisted.internet.error.ConnectionDone'>: Connection was closed cleanly.
2018-11-26 23:10:47+0000 [-] ]
2018-11-26 23:10:47+0000 [-] Stopping factory <__main__.NeteaseMusicProxyClientFactory instance at 0x1093ab3b0>
2018-11-26 23:10:48+0000 [-] http://music.163.com/eapi/copyright/restrict/
2018-11-26 23:10:48+0000 [-] Starting factory <__main__.NeteaseMusicProxyClientFactory instance at 0x10939e170>
2018-11-26 23:10:48+0000 [-] "127.0.0.1" - - [26/Nov/2018:23:10:47 +0000] "POST http://music.163.com/eapi/song/download/limit HTTP/1.1" 200 55 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/605.1.15 (KHTML, like Gecko)"
2018-11-26 23:10:48+0000 [-] [Failure instance: Traceback (failure with no frames): <class 'twisted.internet.error.ConnectionDone'>: Connection was closed cleanly.
2018-11-26 23:10:48+0000 [-] ]
2018-11-26 23:10:48+0000 [-] Stopping factory <__main__.NeteaseMusicProxyClientFactory instance at 0x10939b758>
2018-11-26 23:10:48+0000 [-] "127.0.0.1" - - [26/Nov/2018:23:10:47 +0000] "POST http://music.163.com/eapi/song/download/limit HTTP/1.1" 200 55 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/605.1.15 (KHTML, like Gecko)"
2018-11-26 23:10:48+0000 [-] [Failure instance: Traceback (failure with no frames): <class 'twisted.internet.error.ConnectionDone'>: Connection was closed cleanly.
2018-11-26 23:10:48+0000 [-] ]
2018-11-26 23:10:48+0000 [-] Stopping factory <__main__.NeteaseMusicProxyClientFactory instance at 0x10939b170>
2018-11-26 23:10:48+0000 [-] http://music.163.com/eapi/song/enhance/download/url
2018-11-26 23:10:48+0000 [-] Starting factory <__main__.NeteaseMusicProxyClientFactory instance at 0x1093a7758>
2018-11-26 23:10:48+0000 [-] http://music.163.com/eapi/song/enhance/download/url
2018-11-26 23:10:48+0000 [-] Starting factory <__main__.NeteaseMusicProxyClientFactory instance at 0x1093a1638>
2018-11-26 23:10:48+0000 [-] "127.0.0.1" - - [26/Nov/2018:23:10:48 +0000] "POST http://music.163.com/eapi/copyright/restrict/ HTTP/1.1" 200 63 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/605.1.15 (KHTML, like Gecko)"
2018-11-26 23:10:48+0000 [-] [Failure instance: Traceback (failure with no frames): <class 'twisted.internet.error.ConnectionDone'>: Connection was closed cleanly.
2018-11-26 23:10:48+0000 [-] ]
2018-11-26 23:10:48+0000 [-] Stopping factory <__main__.NeteaseMusicProxyClientFactory instance at 0x10939e170>
2018-11-26 23:10:48+0000 [-] "127.0.0.1" - - [26/Nov/2018:23:10:48 +0000] "POST http://music.163.com/eapi/song/enhance/download/url HTTP/1.1" 200 238 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/605.1.15 (KHTML, like Gecko)"
2018-11-26 23:10:48+0000 [-] [Failure instance: Traceback (failure with no frames): <class 'twisted.internet.error.ConnectionDone'>: Connection was closed cleanly.
2018-11-26 23:10:48+0000 [-] ]
2018-11-26 23:10:48+0000 [-] Stopping factory <__main__.NeteaseMusicProxyClientFactory instance at 0x1093a1638>
2018-11-26 23:10:48+0000 [-] "127.0.0.1" - - [26/Nov/2018:23:10:48 +0000] "POST http://music.163.com/eapi/song/enhance/download/url HTTP/1.1" 200 238 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/605.1.15 (KHTML, like Gecko)"
2018-11-26 23:10:48+0000 [-] [Failure instance: Traceback (failure with no frames): <class 'twisted.internet.error.ConnectionDone'>: Connection was closed cleanly.
2018-11-26 23:10:48+0000 [-] ]
2018-11-26 23:10:48+0000 [-] Stopping factory <__main__.NeteaseMusicProxyClientFactory instance at 0x1093a7758>

**Comment by yi-ji** [comment]
Thanks for reporting!
Is this happening for all songs or few? Are they playable? if only few songs, could you name one so I can test on?

**Comment by yi-ji** [comment]
In fact, this tool has never been tested with music downloading. Seems many songs cannot be downloaded due to url errors. No plan for this feature now, don't wanna be involved in (more severe) copyright issues. 
Some other similar tools replace the download url somehow. More than welcome to any contribution!

In short, sorry! this is going to be marked as "won't fix"

**Comment by xiki-tempula** [comment]
I have tried again today but it appears the probelm has magically fixed itself. However, I agree that it is better to not support the download function.
